iphone/cli/commands/_build.js

```diff
@@ -3132,33 +3137,30 @@ iOSBuilder.prototype.createXcodeProject = function createXcodeProject(next) {
 		},
 		legacySwift = version.lt(this.xcodeEnv.version, '8.0.0');
 
+	this.gccDefs = 'DEPLOYTYPE=' + this.deployType;
 	// set additional build settings
 	if (this.target === 'simulator' || this.target === 'macos') {
-		gccDefs.push('__LOG__ID__=' + this.tiapp.guid);
-		gccDefs.push('DEBUG=1');
-		gccDefs.push('TI_VERSION=' + this.titaniumSdkVersion);
+		this.gccDefs += ' __LOG__ID__=' + this.tiapp.guid;
```

Comment:
Use templated strings, or as suggested above use key/value pairs to stick in a `Map` instead.